app/src/main/java/org/secuso/privacyfriendly2048/helpers/ThemeResolver.kt

```diff
@@ -0,0 +1,43 @@
+package org.secuso.privacyfriendly2048.helpers
+
+import android.content.Context
+import android.preference.PreferenceManager
+import androidx.annotation.ColorInt
+import androidx.annotation.ColorRes
+import androidx.annotation.StyleRes
+import org.secuso.privacyfriendly2048.R
+
+@StyleRes
+fun GetThemeRes(ctx: Context): Int {
+    val prefs = PreferenceManager.getDefaultSharedPreferences(ctx)
+
+    return when (prefs.getString("currentTheme", "system")) {
+        "light" -> R.style.AppThemeLight
+        "dark" -> R.style.AppThemeDark
+        else -> R.style.AppTheme
+    }
+}
+
+@ColorRes
+fun GetColorRes(ctx: Context, res: Int): Int {
+    val prefs = PreferenceManager.getDefaultSharedPreferences(ctx)
+
+    val style: Int
+
+    when (prefs.getString("pref_color", "1")) {
+        "2" -> style = R.style.ColorSchemeOriginal
+        else -> style = R.style.ColorSchemeDefault
+    }
+
+    val ta = ctx.obtainStyledAttributes(style, intArrayOf(res))
+    val resolved = ta.getInt(0, 0)
+    ta.recycle()
+
+    return resolved
+}
+
+@ColorInt
+fun GetColorInt(ctx: Context, res: Int): Int {
```

Comment:
The `setTextColor` requires an actual color instead of a reference like the `GetColorRes` function. Somehow it works if I just cast the reference to an int